examples/cosmos-sdk/transfer.md

```diff
@@ -0,0 +1,243 @@
+- [`poetry`](https://python-poetry.org/docs/#installation): For Atomkraft project
+- [`git`](https://github.com/git-guides/install-git): To clone Atomkraft template and Cosmos-SDK chain repo
+- [`make`](https://www.gnu.org/software/make): Compile Cosmos-SDK chain binary
+- [`go`](https://go.dev/doc/install): Compile Cosmos-SDK chain binary
+- [`java`](https://www.java.com/download): For [Apalache](https://apalache.informal.systems) model checker
+
+#### `ubuntu`
+
+```
+apt update -y && apt upgrade -y
+apt install pip3 git golang curl default-jre -y
+pip install --upgrade poetry
+```
+
+#### `archlinux`
+
+```
+pacman -Syu python-pip python-poetry git make go jre-openjdk gcc --noconfirm --needed
+```
+
+### Install Atomkraft
+
+```
+pip install --upgrade atomkraft
+```
+
+### Initialize project
+
+```
+atomkraft init transfer
+cd transfer
+```
+
+### Turn on Poetry shell
+
+```
+poetry shell
+```
+
+### Create a model specification
+
+We will model a simple token transfer between two users.
+Alice and Bob both start with 100 tokens.
+At each step, Alice or Bob will send some amount of tokens to the other person.
+We will use TLA+ to specify this model.
+You can use the following code to _jump-start_ a new model at `models/transfer.tla`.
+
+```
+cat > models/transfer.tla << EOF
+---- MODULE transfer ----
+EXTENDS Apalache, Variants, Integers, FiniteSets
+
+VARIABLES
+    \* @type: Int -> Int;
+    balances,
+    \* @type: Init({n_wallet: Int}) | Transfer({sender: Int, receiver: Int, amount: Int});
+    action,
+    \* @type: Int;
+    step
+
+WALLETS == 0..1
+
+Init ==
+    /\ balances = [wallet \in WALLETS |-> 100]
+    /\ action = Variant("Init", [n_wallet |-> Cardinality(WALLETS)])
+    /\ step = 0
+
+Next ==
+    \E sender \in WALLETS:
+    \E receiver \in WALLETS:
+    \E amount \in 0..balances[sender]:
+        /\ sender /= receiver
+        /\ balances' = [
+            balances EXCEPT
+            ![sender] = @ - amount,
+            ![receiver] = @ + amount
+            ]
+        /\ action' = Variant("Transfer", [sender |-> sender, receiver |-> receiver, amount |-> amount])
+    /\ step' = step + 1
+
+View ==
+    IF VariantTag(action) = "Transfer"
+    THEN VariantGetUnsafe("Transfer", action)
+    ELSE [sender |-> -1, receiver |-> -1, amount |-> 0]
+
+Inv == step < 10
+
+====
+EOF
+```
+
+### Generate traces
+
+#### Download Apalache
+
+We will use Apalache to generate traces from our model.
+
+```
+curl -Lo- https://github.com/informalsystems/apalache/releases/download/v0.26.0/apalache-0.26.0.tgz | tar -zxf-
+```
+
+The Apalache executable will be at `./apalache-0.26.0/bin/apalache-mc`.
+
+The following will generate traces at `traces/`.
+
+```
+./apalache-0.26.0/bin/apalache-mc check --features=rows --init=Init --next=Next --inv=Inv --view=View --max-error=10 --run-dir=mc_traces models/transfer.tla
+ls -1I violation.itf.json mc_traces/violation*itf.json | xargs cp -t traces
```

Comment:
I get two errors with this command (in macos, bash). First `ls: violation.itf.json: No such file or directory`. Then, if I remove `violation.itf.json` and run `ls -1I mc_traces/violation*itf.json | xargs cp -t traces`, I get a `cp: illegal option -- t`. Apparently the `-t` option is a GNU extension and not available in macos. What it works for me is `find . -iname "violation*itf.json" -exec cp -- "{}" traces \;`.

Reply:
(In fact, if we were able to run `atomkraft model check ...` instead of apalache-mc, we could specify the target directory for the traces, and we wouldn't need all these file handling.)

Reply:
I want to exclude `violation.itf.json`. I will update `xargs` command as `xargs -I _ cp _ traces`.

Reply:
> if we were able to run `atomkraft model check ...`

Yes. once that is ready, I will update the tutorial/e2e test :slightly_smiling_face: